### PR TITLE
Fixed the collapsing of hunks

### LINKED ISSUE #169: Collapsing of diff hunks is broken
When you click on the collapse button, the diff is not collapsed.

### PR DESCRIPTION
**Pull request by eanker**
Fixed #169 

The state is still shared between childs, but that is the same for the whole group and comments.
This is something that should be fixed, but I believe it's out of scope of this issue.
The issue is reported at the polymer repo, namely 3505.
But the proposed solution does not seem to work.
`When initializing a property to an object or array value, use a function to ensure that each element gets its own copy of the value, rather than having an object or array shared across all instances of the element.`

Edit:
This issue is now fixed by the second commit of this pull request.
The collapsed property is reset on every reload.

### CODE REVIEW

File: src/projects/diff/group.html [hunk 1]
```diff
@@ -180,6 +180,7 @@ <h2 hidden$="[[editable]]">[[displayTitle]]</h2>
         },
         project: Object
       },
+      observers: ['_resetCollapsed(group)'],
```
Comment (TimvdLippe) [inline src/projects/diff/group.html:183]: :tada:

File: src/projects/diff/group.html [hunk 2]
```diff
@@ -165,7 +165,7 @@ <h2 hidden$="[[editable]]">[[displayTitle]]</h2>
           value: ''
         },
         collapseComments: {
-          type: Boolean,
+          type: Object,
```
Comment (TimvdLippe) [inline src/projects/diff/group.html]: I think this still should be a `Boolean`?

**Comment by TimvdLippe** [comment]
To prevent regression, maybe we can add a small test that verifies that the display style of the highlight is `none` when the content is collapsed. Other than the small comment LGTM!

**Comment by txsmith** [comment]
Confirmed working!
